File: src/ai/Embedding.NestJS/embedding-options.ts
export type EmbeddingOptions = {
  ollamaBaseUrl: string;
  model: string;
  timeoutMs: number;
};

export const EMBEDDING_OPTIONS = Symbol('EMBEDDING_OPTIONS');

export function createEmbeddingOptions(): EmbeddingOptions {
  return {
    ollamaBaseUrl: process.env.OLLAMA_BASE_URL || process.env.OLLAMA_EMBEDDING_BASE_URL || 'http://localhost:11434',
    model: process.env.OLLAMA_EMBEDDING_MODEL || 'nomic-embed-text',
    timeoutMs: Number(process.env.OLLAMA_EMBEDDING_TIMEOUT_MS || 60000),
  };
}

File: src/ai/Embedding.DotNet/EmbeddingOptions.cs
namespace Embedding.DotNet;

public sealed class EmbeddingOptions
{
    public string OllamaBaseUrl { get; set; } = "http://localhost:11434";
    public string Model { get; set; } = "nomic-embed-text";
    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(60);
}